doc/rt_control.md

```diff
@@ -0,0 +1,142 @@
+<!--
+SPDX-FileCopyrightText: 2025, Yaskawa America, Inc.
+SPDX-FileCopyrightText: 2025, Delft University of Technology
+
+SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# R/T Motion Control
+
+The real-time motion control server is intended to be used in a closed loop system.
+It allows the user to command incremental offsets at the rate of the robot controller's interpolation clock.
+This control mode minimizes overhead as much as possible by routing the user commands directly to the MotoPlus motion API.
+
+## Activation
+
+This control mode is activated using the [StartRtMode](ros_api.md#start_rt_mode) service.
+The user must specify the `control_mode` to indicate whether the increments will be joint offsets (radians) or cartesian TCP offsets (meters / radians).
+
+If this service is successful, it will return a `result_code` of `Ready (1)`.
+Otherwise, please examine the `result_code` and `message` for more information.
+
+The service will also return a `period` in milliseconds.
+This indicates the rate at which increment commands will be expected by the robot.
+The default `period` for a single robot arm is 4 milliseconds.
+However, that value will increase as additional axes or arms are added to the system.
+
+## Usage
+
+### Command Flow
+
+Once activated, a UDP server will listen on port `8889` (default).
+The user then sends the first increment with a `sequenceId = 0`.
+After that, the user must wait until the robot replies before sending the next increment.
+Each subsequent command must increment the `sequenceId`. Additionally, each subsequent command must not be sent until the robot replies to the previous command.
+This will occur at the rate of the `period` from the [StartRtMode](ros_api.md#start_rt_mode) service.
+
+If a command is not received with 30 seconds (default), then the session times out and is dropped.
+At that point, the server must be reactivated by calling `stop_traj_mode` and `start_rt_mode`.
+A "keep-alive" can be used by sending a command with zero increments.
+
+Additionally, if the client does not receive a reply packet within this amount of time, then it should be assumed that the session is dead.
+
+<img src="img/RtFlow.png" alt="Command Flow" />
+
+### Data format (command)
+
+The command packet is a *packed* `RtPacket` structure.
+This contains a sequence ID, the increments for each control group, and the tool number to use for each control group.
+
+```c
+//##########################################################################
+//                  !All data is little-endian!
+//##########################################################################
+struct RtPacket
```

Comment:
should this perhaps also include a bitmask/integer array indicating to which groups increments are being commanded?

That would make it an even thinner wrapper around `mpExRcsIncrementMove(..)` while making it possible to *not* have to send `0` increments for groups that *should not move*.